wasp/src/main/java/com/orhanobut/wasp/WaspResponse.java

```diff
@@ -54,4 +54,8 @@ public void log(LogLevel logLevel) {
                 // Method is called but log level is not meant to log anything
         }
     }
+
+    public WaspError toWaspError(String errorMsg) {
+        return new WaspError(url, statusCode, headers, errorMsg, body.getBytes(), networkTime);
```

Comment:
<code>body</code> might be null, I would suggest a null control here. 
<code>
byte[] bodyBytes = body != null ? body.getBytes() : null;
</code>